src/components/views/rooms/RoomTile.tsx

```diff
@@ -81,161 +61,63 @@ const contextMenuBelow = (elementRect: PartialDOMRect) => {
-                MessagePreviewStore.getPreviewChangedEventName(this.props.room),
-                this.onRoomPreviewChanged,
-            );
-            CommunityPrototypeStore.instance.off(
-                CommunityPrototypeStore.getUpdateEventName(prevProps.room?.roomId),
-                this.onCommunityUpdate,
-            );
-            CommunityPrototypeStore.instance.on(
-                CommunityPrototypeStore.getUpdateEventName(this.props.room?.roomId),
-                this.onCommunityUpdate,
-            );
-            prevProps.room?.off("Room.name", this.onRoomNameUpdate);
-            this.props.room?.on("Room.name", this.onRoomNameUpdate);
-        }
     }
 
     public componentDidMount() {
         // when we're first rendered (or our sublist is expanded) make sure we are visible if we're active
-        if (this.state.selected) {
+        if (this.props.model.selected) {
             this.scrollIntoView();
         }
+        this.props.model.on("change", this.rerender);
+        this.props.model.on("ensureVisible", this.scrollIntoView);
     }
 
-    public componentWillUnmount() {
-        if (this.props.room) {
-            ActiveRoomObserver.removeListener(this.props.room.roomId, this.onActiveRoomUpdate);
-            MessagePreviewStore.instance.off(
-                MessagePreviewStore.getPreviewChangedEventName(this.props.room),
-                this.onRoomPreviewChanged,
-            );
-            CommunityPrototypeStore.instance.off(
-                CommunityPrototypeStore.getUpdateEventName(this.props.room.roomId),
-                this.onCommunityUpdate,
-            );
-            this.props.room.off("Room.name", this.onRoomNameUpdate);
+    public componentDidUpdate(prevProps: Readonly<IProps>, prevState: Readonly<IState>) {
+        if (prevProps.model !== this.props.model) {
+            prevProps.model.off("change", this.rerender);
+            prevProps.model.off("ensureVisible", this.scrollIntoView);
+            prevProps.model.destroy();
+            this.props.model.on("change", this.rerender);
+            this.props.model.on("ensureVisible", this.scrollIntoView);
         }
-        defaultDispatcher.unregister(this.dispatcherRef);
-        this.notificationState.off(NOTIFICATION_STATE_UPDATE, this.onNotificationUpdate);
     }
 
-    private onAction = (payload: ActionPayload) => {
-        if (payload.action === "view_room" && payload.room_id === this.props.room.roomId && payload.show_room_tile) {
-            setImmediate(() => {
-                this.scrollIntoView();
-            });
-        }
-    };
-
-    private onCommunityUpdate = (roomId: string) => {
-        if (roomId !== this.props.room.roomId) return;
-        this.forceUpdate(); // we don't have anything to actually update
-    };
+    public componentWillUnmount() {
+        this.props.model.off("change", this.rerender);
+        this.props.model.off("ensureVisible", this.scrollIntoView);
+        this.props.model.destroy();
```

Comment:
it's a little surprising that the model gets destroyed here, but I suppose it's the right thing given not much else will know when is best to destroy it...

Reply:
Indeed, in a larger example, the parent view model would destroy the child view model when needed (e.g. upon navigation, ...), and the child view model being cleared would cause a re-render and the component to unmount. Given there is no parent view model here yet, I reversed things.